packages/core/src/command/consolidated/consolidated-filter-by-date.ts

```diff
@@ -0,0 +1,334 @@
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementRange) {
+    const res = areRangesWithinRange([resource.abatementRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementString) {
+    const res = isDateWithinDateRange(resource.abatementString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementDateTime) {
+    const res = isDateWithinDateRange(resource.abatementDateTime, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isEncounterWithinDateRange(resource: Encounter, range: DateRange): boolean {
+  if (resource.period) {
+    const res = arePeriodsWithinRange([resource.period], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.location) {
+    const periods = resource.location.map(l => l.period);
+    const res = arePeriodsWithinRange(periods, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isFamilyMemberHistoryWithinDateRange(
+  resource: FamilyMemberHistory,
+  range: DateRange
+): boolean {
+  if (resource.condition) {
+    const periods = resource.condition.map(c => c.onsetPeriod);
+    const arePeriods = arePeriodsWithinRange(periods, range);
+    if (arePeriods !== undefined) return arePeriods;
+
+    const ranges = resource.condition.map(c => c.onsetRange);
+    const areRanges = areRangesWithinRange(ranges, range);
+    if (areRanges !== undefined) return areRanges;
+
+    const dates = resource.condition.flatMap(c => safeDate(c.onsetString) ?? []);
+    const areDates = areDatesWithinRange(dates, range);
+    if (areDates !== undefined) return areDates;
+  }
+  if (resource.date) {
+    const res = areDatesWithinRange([resource.date], range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isDiagnosticReportWithinDateRange(resource: DiagnosticReport, range: DateRange): boolean {
+  const dates = getDatesFromEffectiveDateTimeOrPeriod(resource);
+  const res = areDatesWithinRange(dates, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isDocumentReferenceWithinDateRange(
+  resource: DocumentReference,
+  range: DateRange
+): boolean {
+  const res = isPeriodWithinRange(resource.context?.period, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isConsentWithinDateRange(resource: Consent, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.dateTime, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isCoverageWithinDateRange(resource: Coverage, range: DateRange): boolean {
+  const res = isPeriodWithinRange(resource.period, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isGoalWithinDateRange(resource: Goal, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.startDate, range);
```

Comment:
missed resource.target.dueDate (we have this one in converter) + 
resource.statusDate